### PR TITLE
CGAL with --ipelets

### PR DESCRIPTION
**Pull request by mrzv**
Added `--ipelets` option to `cgal` formula + fix the `IPEPREFIX` path in `ipe` formula.

### CODE REVIEW

File: Library/Formula/cgal.rb [hunk 1]
```diff
@@ -51,6 +56,16 @@ def install
     args << '.'
     system "cmake", *args
     system "make install"
+
+    if build.with? "ipelets"
+      cd "demo/CGAL_ipelets"
+      system "cmake . -DCGAL_DIR=../.."
+      system "make"
+      system "make install DESTDIR=#{prefix}"
```
Comment (mikemcquaid) [inline Library/Formula/cgal.rb]: Can you split this into `system "make", "install", "DESTDIR=#{prefix}"`

File: Library/Formula/cgal.rb [hunk 2]
```diff
@@ -51,6 +56,16 @@ def install
     args << '.'
     system "cmake", *args
     system "make install"
+
+    if build.with? "ipelets"
+      cd "demo/CGAL_ipelets"
```
Comment (mikemcquaid) [inline Library/Formula/cgal.rb]: Put the below in a `cd "demo/CGAL_ipelets" do` block.

File: Library/Formula/ipe.rb [hunk 3]
```diff
@@ -38,7 +38,7 @@ class Ipe < Formula
 
   def install
     cd 'src' do
-      system "make", "IPEPREFIX=#{prefix}"
+      system "make", "IPEPREFIX=#{HOMEBREW_PREFIX}"
```
Comment (mikemcquaid) [inline Library/Formula/ipe.rb]: What's the motivation for this change, out of interest?

File: Library/Formula/cgal.rb (hunk 1 above)
Comment (mikemcquaid) [inline Library/Formula/cgal.rb]: Also, maybe work installing this into a location in the `buildpath` and then using `lib.install` from there rather than putting it in the `prefix` and then deleting it.

File: Library/Formula/cgal.rb [hunk 4]
```diff
@@ -51,6 +56,16 @@ def install
...
+      system "cmake . -DCGAL_DIR=../.."
```
Comment (mikemcquaid) [inline Library/Formula/cgal.rb]: `system "cmake", ".", "-DCGAL_DIR=../.."`

File: Library/Formula/cgal.rb [hunk 5]
```diff
@@ -13,9 +13,12 @@ class Cgal < Formula
 
   option :cxx11
 
-  option 'imaging', "Build ImageIO and QT compoments of CGAL"
+  deprecated_option "imaging" => "with-imaging"
```
Comment (mikemcquaid) [inline Library/Formula/cgal.rb]: :+1:

**Comment by mikemcquaid** [comment]
A few tweaks and needs rebased on `master`.

File: Library/Formula/ipe.rb (hunk 3 above)
Comment (mrzv) [inline Library/Formula/ipe.rb]: That's the main change of the pull request. It fixes a bug in how homebrew installs Ipe, so that other packages could install ipelets. The other change of this pull request adds CGAL's ipelets, but that's an enhancement. They could be installed as a separate package.

**Comment by mrzv** [comment]
To be honest, my desire to deal with this request (which itself is a refactoring of an earlier request) has been exhausted. All the minor things you list, you could fix yourself in less time than it took you to write them as comments.

**Comment by mikemcquaid** [comment]
This PR needs rebased anyway and the last build that ran failed. Maybe I could do it in less time but Homebrew has hundreds of open pull requests and that doesn't scale. Closing this if you don't want to work on this any more.

File: Library/Formula/ipe.rb (hunk 3 above)
Comment (DomT4) [inline Library/Formula/ipe.rb]: FWIW, fixed this bug for you the other day: https://github.com/Homebrew/homebrew/blob/master/Library/Formula/ipe.rb#L47-L48

Comment (mrzv) [inline Library/Formula/ipe.rb]: Yeah, I noticed it after yesterday's exchange. Thanks for doing it, Dominyk. You've effectively solved the big problem. I now simply install CGAL ipelets manually.

Comment (DomT4) [inline Library/Formula/ipe.rb]: No worries. Bumping the version allowed me to fix a few of the other bugs that were lying around the formula, which was good. Upstream have said they will never officially support ` ipe ` for OS X, which is deeply frustrating, but as long as it works, we're happy to keep carrying it.
